app/core/models.py

```diff
@@ -383,3 +383,12 @@ class CheckType(AbstractBaseModel):
 
     def __str__(self):
         return f"{self.name}"
+
+
+class ProjectProgramAreaXref(AbstractBaseModel):
+    project_id = models.ForeignKey(Project, on_delete=models.CASCADE)
+    program_area_id = models.ForeignKey(ProgramArea, on_delete=models.CASCADE)
+    created_date = models.DateTimeField("Created at", null=True, blank=True)
```

Comment:
The inherited AbstractBaseModel class already creates a timestamp for creation; see [here](https://github.com/del9ra/peopledepot/blob/9465c5100a9e45c05ad72273fcf22e6839b415a1/app/core/models.py#L12)

Reply:
Why did they use the DateTimeField in the Add the Model section of this guide: https://hackforla.github.io/peopledepot/how-to/add-model-and-api-endpoints/, and in the Project model as "completed_at", in the Event model as "start_time", and in the Affiliation model as "ended_at"?

Reply:
@del9ra there's some translation required between the DB notation in the ERD and spreadsheet and the Django data fields. We use an abstract base class (ABC) so that all the models will have create and modify timestamps. In the database design, many but not all models have the timestamps. We decided to make that a standard thing at one point.

So you should ignore anything from the database table that does the same as create and update timestamps.

At the time, I was trying to decide what to call these standard timestamps, and I found some page which discussed "on" vs "at", where "on" is usually a date and "at" is usually a time. So I added "_at" to the end of the names. But eventually I found a similar `TimeStampedModel` ABC in the widely-used django package [django-extensions](https://django-extensions.readthedocs.io/en/latest/model_extensions.html#database-model-extensions). It would've been a good idea to use that too.

Reply:
Hmm... looks like this xref table doesn't have any extra data fields other than the relations (foreign keys). It means that we don't really need to create a Django model for this table at all. I'll have to go though the other issues and pick out similar issues before someone works on them.

Reply:
@del9ra I'm sorry, but I need to change this issue from "create xref table" to "create many to many relation between Project and ProgramArea models". We don't have any existing code in the repository to serve as an example, but I can find some other examples if you want to work on it.

Reply:
> Why did they use the DateTimeField in the Add the Model section of this guide: https://hackforla.github.io/peopledepot/how-to/add-model-and-api-endpoints/, and in the Project model as "completed_at", in the Event model as "start_time", and in the Affiliation model as "ended_at"?

@del9ra Did you get your question answered? I'm not sure why those were included either. This should be added to the meeting agenda if we can't resolve this. I'm not sure of how the decisions were made for completed, start, ended and so on. Looking at the ERD also shows that completed_at has a type of "date" versus "DateTimeField". It looks like the ERD also needs to be updated for at least Project, if not all the other tables.

Reply:
> @del9ra there's some translation required between the DB notation in the ERD and spreadsheet and the Django data fields. We use an abstract base class (ABC) so that all the models will have create and modify timestamps. In the database design, many but not all models have the timestamps. We decided to make that a standard thing at one point.
> 
> So you should ignore anything from the database table that does the same as create and update timestamps.
> 
> At the time, I was trying to decide what to call these standard timestamps, and I found some page which discussed "on" vs "at", where "on" is usually a date and "at" is usually a time. So I added "_at" to the end of the names. But eventually I found a similar `TimeStampedModel` ABC in the widely-used django package [django-extensions](https://django-extensions.readthedocs.io/en/latest/model_extensions.html#database-model-extensions). It would've been a good idea to use that too.

@fyliu I noticed that we have the `ended_at` field in the Affiliation model; `ended` in Permission model, project_sponsor_partner_xref, permission_history; `ended_on` in project_language_xref and project_sdg_xref. How does this work in this case? Do we need to create a single field name for them in the ABC model?

Reply:
@del9ra we could. It looks like the `ended` field applies to xref tables where the relationships ended but we're keeping the row around (soft delete). But we don't have a lot of those tables to apply it to, so we shouldn't do this for all the models. Then again, you pointed out that the few are already inconsistent in naming, so maybe we should do something to make/keep them consistent.

I'm thinking we should bring up the inconsistent naming to @ExperimentsInHonesty and @Neecolaa and at least make an issue (ER maybe) to fix it on some tables. 

We should bring this to the meeting and decide on whether or not to make an ABC model or something else to help force this consistency.